crates/store/src/db/sql/mod.rs

```diff
@@ -119,6 +121,8 @@ pub fn select_accounts_by_block_range(
     block_end: BlockNumber,
     account_ids: &[AccountId],
 ) -> Result<Vec<AccountSummary>> {
+    QueryParamAccountIdLimit::check(account_ids.len())?;
```

Comment:
Should we check these limits at the API level (i.e., even before we get to making database requests)? Also, it may make sense to impose these limits in the RPC component as well - so that invalid requests don't make it to the store component at all.

Reply:
My intent was to keep the actual query closeby the limitation. In an error case it would mean additional unwind steps, but the commoncase wouldn't be hurt.
Are we planning to expose the store / its RPC API publicly? If so, that'd be an argument to move it as close as possible to the RPC entrypoints for DoS mitigations.

Reply:
In that case I pull out the logic and re-use for both.

Reply:
Pulled the logic into `limiter.rs` and pushed it to the query edges just after we parse the tonic input.

Reply:
I think the primary checks should be in the RPC component (i.e., [here](https://github.com/0xMiden/miden-node/blob/de0e4caffc779867235036361aaec6acecfd87c0/crates/rpc/src/server/api.rs)). And the secondary checks could be at the DB level. We could have a third set of checks at the store's RPC API (as you have it now) - but I'm not sure that's necessary.

Reply:
I clearly misunderstood that.